### ISSUE TITLE
Cloning (Help ELI5) 

### ISSUE THREAD

**Issue by dildosven** [AlexCSDev/PatreonDownloader#226]
Can someone help me explain this section, cant figure it out. When I run "git clone --recurse-submodules" in the main folder I just get "fatal: You must specify a repository to clone."

## IMPORTANT: Cloning PatreonDownloader properly
You need to clone submodules in order to build PatreonDownloader.
If you have not yet cloned the repository use **git clone --recurse-submodules**.
If you have cloned the repository without submodules use **git submodule update --init --recursive**.

**Comment by dildosven** [comment]
NVM
"git clone --recurse-submodules https://github.com/AlexCSDev/PatreonDownloader"

**Comment by dildosven** [comment]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.PuppeteerEngine\UniversalDownloaderPlatform.PuppeteerEngine.csproj : error NU1101: Unable to find package Ninject. No packages exist with this id in source(s): Microsoft
Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.Implementation\PatreonDownloader.Implementation.csproj : error NU1101: Unable to find package HtmlAgilityPack. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\Patre
onDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.Engine\UniversalDownloaderPlatform.Engine.csproj : error NU1101: Unable to find package HtmlAgilityPack. No packages exist with this id in source(s): Microsoft Visual Stu
dio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\UniversalDownloaderPlatform.DefaultImplementations.csproj : error NU1101: Unable to find package HtmlAgilityPack. No packages exist with this id in
 source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.Implementation\PatreonDownloader.Implementation.csproj : error NU1101: Unable to find package MimeTypesMap. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonD
ownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.Engine\UniversalDownloaderPlatform.Engine.csproj : error NU1101: Unable to find package Microsoft.Extensions.Configuration.Json. No packages exist with this id in source(
s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.Implementation\PatreonDownloader.Implementation.csproj : error NU1101: Unable to find package Ninject. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownlo
ader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package CommandLineParser. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\UniversalDownloaderPlatform.DefaultImplementations.csproj : error NU1101: Unable to find package Ninject. No packages exist with this id in source(
s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.Engine\UniversalDownloaderPlatform.Engine.csproj : error NU1101: Unable to find package Ninject. No packages exist with this id in source(s): Microsoft Visual Studio Offl
ine Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.Implementation\PatreonDownloader.Implementation.csproj : error NU1101: Unable to find package NLog. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloade
r\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.Engine\UniversalDownloaderPlatform.Engine.csproj : error NU1101: Unable to find package NLog. No packages exist with this id in source(s): Microsoft Visual Studio Offline
 Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\UniversalDownloaderPlatform.DefaultImplementations.csproj : error NU1101: Unable to find package NLog. No packages exist with this id in source(s):
 Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.Implementation\PatreonDownloader.Implementation.csproj : error NU1101: Unable to find package ninject.extensions.conventions. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\PAD
A\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.Implementation\PatreonDownloader.Implementation.csproj : error NU1101: Unable to find package PuppeteerSharp. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\Patreo
nDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package NLog. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.PuppeteerEngine\UniversalDownloaderPlatform.PuppeteerEngine.csproj : error NU1101: Unable to find package ninject.extensions.conventions. No packages exist with this id i
n source(s): Microsoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package Microsoft.NET.ILLink.Tasks. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package Microsoft.NET.ILLink.Analyzers. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package HtmlAgilityPack. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package MimeTypesMap. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.PuppeteerEngine\UniversalDownloaderPlatform.PuppeteerEngine.csproj : error NU1101: Unable to find package NLog. No packages exist with this id in source(s): Microsoft Vis
ual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.PuppeteerEngine\UniversalDownloaderPlatform.PuppeteerEngine.csproj : error NU1101: Unable to find package PuppeteerSharp. No packages exist with this id in source(s): Mic
rosoft Visual Studio Offline Packages [D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj]
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package Ninject. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package Microsoft.Extensions.Configuration.Json. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package ninject.extensions.conventions. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
D:\PADA\Patreon\PatreonDownloader\PatreonDownloader.App\PatreonDownloader.App.csproj : error NU1101: Unable to find package PuppeteerSharp. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages

The build failed. Fix the build errors and run again.

Im guessing I'm doing something wrong cloning the submodules

**Comment by martin9020** [comment]
Everyone is getting these errors thats why these programers are trying to be smart and not explain the whole picture. Firstly you have to update all packages. But then you go to this error Severity	Code	Description	Project	File	Line	Suppression State
Warning (active)	CS0168	The variable 'ex' is declared but never used	UniversalDownloaderPlatform.DefaultImplementations	C:\Users\marti\Source\Repos\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\WebDownloader.cs	328	
Warning (active)	CS1998	This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.	UniversalDownloaderPlatform.DefaultImplementations	C:\Users\marti\Source\Repos\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\RemoteFileSizeChecker.cs	22	
Warning (active)	CS0168	The variable 'ex' is declared but never used	UniversalDownloaderPlatform.DefaultImplementations	C:\Users\marti\Source\Repos\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\WebDownloader.cs	279	
Warning (active)	CS0168	The variable 'ex' is declared but never used	UniversalDownloaderPlatform.DefaultImplementations	C:\Users\marti\Source\Repos\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.DefaultImplementations\WebDownloader.cs	400	
Warning (active)	CS1998	This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.	UniversalDownloaderPlatform.Engine	C:\Users\marti\Source\Repos\PatreonDownloader\submodules\UniversalDownloaderPlatform\UniversalDownloaderPlatform.Engine\Helpers\URLChecker.cs	15	

And then you realize the code is broken and have to go into the code and do some stuff
